seeq/addons/correlation/utils/_sdl.py

```diff
@@ -71,3 +72,40 @@ def get_workbook_worksheet_workstep_ids(url):
     if 'workstepId' in params:
         workstep_id = params['workstepId'][0]
     return workbook_id, worksheet_id, workstep_id
+
+
+def is_ipv4(string):
+    temp_str = None
+    try:
+        temp_str = string.split("https://")[1]
+    except IndexError:
+        try:
+            temp_str = string.split("http://")[1]
+        except IndexError:
+            print('Unrecognisable URL or IP returned by SPy')
+
+    try:
+        ipaddress.IPv4Network(temp_str)
+        return True
+    except ValueError:
+        return False
+
+
+def get_seeq_url():
+    url_ = None
+    if hasattr(spy.session, 'public_url'):
+        if not is_ipv4(spy.session.public_url):
+            url_ = spy.session.public_url
+        else:
+            if hasattr(spy.session, 'private_url'):
+                if not is_ipv4(spy.session.private_url):
+                    url_ = spy.session.private_url
+                else:
+                    url_ = input(f"\n Please Input Seeq base URL (eg: https://example.seeq.site): ")
+            else:
+                url_ = input(f"\n Please Input Seeq base URL (eg:https://example.seeq.site): ")
+
+    if url_ is not None:
+        return url_
+    else:
+        print("ERROR Obtaining valid URl")
```

Comment:
Same as my comment above. If there is an error here, an exception should be raised. Otherwise the function will continue and the script will failed somewhere else (ungracefully)

Reply:
With the improved logic add, I cant see a path to return a None any more. Only to test and rule out each spy.session functions working and then asking the user for the base URL. 

Would you suggest a method to check the URL supplied by the user is at least active/viewable, like try/rise exception for  `if urllib.request.urlopen("https://example.seeq.com").getcode()) == 200`to have a method to fail gracefully in the case the user inputs its incorrectly?
Could fail for on-prem installs though. Thoughts?

Reply:
I don't think it is needed.